sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor.cs

```diff
@@ -1768,34 +1772,51 @@ private async Task RunProcessingAsync(CancellationToken cancellationToken)
                 }
             }
 
-            // Some ownership for some previously claimed partitions may have expired or have been stolen; stop the processing for partitions
-            // which are no longer owned.
+            // Some ownership for some previously claimed partitions may have expired or have been stolen; stop the processing
+            // for partitions which are no longer owned.
 
             cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+            var ownedPartitions = new HashSet<string>(LoadBalancer.OwnedPartitionIds);
```

Comment:
I looked at changing the load balancer implementation to use a hash set, but it would need to be a new member that would incur roughly the same allocation pattern per cycle, as the partitions are managed in a concurrent dictionary and this is just the `Keys` property being surfaced.

Reply:
Had a thought and worked around this in the load balancer.